test/e2e/apimachinery/aggregator.go

```diff
@@ -145,7 +117,11 @@ func TestSampleAPIServer(f *framework.Framework, aggrclient *aggregatorclient.Cl
 		},
 	}
 	_, err := client.CoreV1().Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{})
-	framework.ExpectNoError(err, "creating secret %q in namespace %q", secretName, namespace)
+	framework.ExpectNoError(err, "creating secret %s in namespace %s", secretName, namespace)
+	defer func() {
```

Comment:
All these defer are not needed, the framework cleans all the recourcea created

Reply:
Oops, missed this one. Framework will clean up after the test has ended, right? As everything is hanging from the same namespace. If there is another test added to this file it will be polluted with resources from this one, should we not clean up explicitly?

Reply:
scratch my comment, it seems you need the defers because those were part of the `cleanTest` function

Reply:
everything in the test namespace gets deleted automatically and will not conflict between tests

cluster-scoped things and things in kube-system need explicit cleanup